framework/platform/bpl/uci/cfg/bpl_cfg.cpp

```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
 
 int cfg_get_wifi_params(const char iface[BPL_IFNAME_LEN], struct BPL_WLAN_PARAMS *wlan_params)
 {
-    int retVal = 0;
-    int index  = 0;
-
     if (!iface || !wlan_params) {
+        MAPF_ERR("cfg_get_wifi_params: invalid input: iface = "
+                 << intptr_t(iface) << " wlan_params = " << intptr_t(wlan_params));
         return RETURN_ERR;
     }
 
-    retVal = cfg_get_index_from_interface(iface, &index);
-    if (retVal) {
-        return retVal;
-    }
+    char ifname[BPL_IFNAME_LEN] = {0};
+    utils::copy_string(ifname, iface, BPL_IFNAME_LEN);
```

Comment:
Why is this copied instead of just using `iface.c_str()`?

Reply:
although I changed to std:string as you requested, the BPL UCI APIs implementation requires non-const char arrays as changes might be made inside the UCI lib to the provided string.

Reply:
Indeed, uci_lookup_ptr modifies its path argument. But it would be more appropriate to do the copy in uci_cfg_get() then. Anyway, something for a separate PR.

BTW, I believe cfg_uci_get_wireless() is no longer used and can be removed.